eks/cleanup.go

```diff
@@ -127,7 +127,12 @@ func deleteDependencies(ec2Svc *ec2.EC2, securityGroupID string) error {
 			AttachmentId: ni.Attachment.AttachmentId,
 		}
 		_, err := ec2Svc.DetachNetworkInterface(detachInput)
+
+		// If we have an error, check that it's NotFound. This is actually a success.
 		if err != nil {
+			if awsErr, isAwsErr := err.(awserr.Error); isAwsErr && awsErr.Code() == "InvalidAttachmentID.NotFound" {
+				logger.Infof("Network interface %s is detached.", aws.StringValue(ni.NetworkInterfaceId))
+			}
```

Comment:
We should return nil here.

Reply:
Reminder: this should still return `nil` in the inner if block to avoid returning the error.

Reply:
No, it cannot return here.

Reply:
I see what you're saying, though. Let me formulate my suggestion...

Reply:
I believe adding a `continue` will avoid returning that error on 143. We cannot return here, however. We must continue the loop.

Reply:
Okay, if I understand correctly, we actually need a break label here. Making that change now.

Reply:
This comment thread is out of date! I was confused, things were confusing, and it has been rewritten to be clearer (hopefully).